src/main/java/com/aws/greengrass/logmanager/model/LogFileGroup.java

```diff
@@ -95,15 +119,16 @@ public static LogFileGroup create(ComponentLogConfiguration componentLogConfigur
             logger.atDebug().kv("component", componentName)
                     .kv("directory", directoryURI)
                     .log("No component logs are found in the directory");
-            return new LogFileGroup(Collections.emptyList(), filePattern, new HashMap<>(), false, Optional.empty());
+            return new LogFileGroup(
+                    Collections.emptyList(), filePattern, new HashMap<>(), false,
+                    lastProcessed,Optional.empty());
         }
 
         boolean isUsingHardlinks = true;
         List<LogFile> logFiles;
 
         files = Arrays.stream(files)
                 .filter(File::isFile)
-                .filter(file -> lastUpdated.isBefore(Instant.ofEpochMilli(file.lastModified())))
```

Comment:
Would not this create hardlinks for the already processed files as well?

Reply:
That is a valid point but I believe we might want that. Already processed files can rotate as well and tracking them for deletion will ensure we always delete the correct file even if a rotation happens